scm/git.go

```diff
@@ -13,6 +13,12 @@ import (
 const EmptyCommit = "0000000000000000000000000000000000000000"
 
 type (
+	// CreateBranch provides a SHA for creating a branch.
+	CreateBranch struct {
+		Ref string
```

Comment:
just a question ... the branch will always have `refs/heads/` in the prefix.  Would it make sense to accept branch name, and then it would be the responsibility of the driver to determine whether or not to prefix with `refs/heads/`?  Similar to how find branch takes a branch name, as opposed to a full ref, and the underlying driver has to decide whether or not to prefix.